### PR TITLE
Add GCE provider for ManageIQ

### PR DESCRIPTION
**Pull request by snecklifter**
##### SUMMARY
This commit adds support for creating GCE providers on ManageIQ.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
remote_management/manageiq

##### ANSIBLE VERSION
```
ansible 2.5.0 (add-gce-provider 79fa4f9e01) last updated 2018/02/04 15:29:19 (GMT +100)
```

##### ADDITIONAL INFORMATION
Hoping to get this final public cloud provider merged before 2.5 window closes.
This adds another parameter so maybe the refactor can happen in 2.6?
Hostname is required by the module but not by manageiq so not sure if we should look at dropping that requirement in the code.

### CODE REVIEW

File: lib/ansible/modules/remote_management/manageiq/manageiq_provider.py [hunk 1]
```diff
@@ -535,6 +540,19 @@
...
       userid: admin
       password: password
+
+
+- name: Create a new GCE provider in ManageIQ
+  manageiq_provider:
+    name: 'EngGoogle'
+    type: 'GCE'
+    provider_region: 'europe-west1'
+    google_project: 'project1'
```
Comment (yaacov) [inline lib/ansible/modules/remote_management/manageiq/manageiq_provider.py]: google_project ?

Comment (snecklifter) [reply lib/ansible/modules/remote_management/manageiq/manageiq_provider.py]: doh! copy/paste/pasta error, thanks.

**Comment by yaacov** [comment]
shipit

File: lib/ansible/modules/remote_management/manageiq/manageiq_provider.py [hunk 2]
```diff
@@ -61,6 +61,11 @@
     default: null
     description: Microsoft Azure subscription ID. defaults to None.
     version_added: "2.5"
+  project:
+    required: false
+    default: null
+    description: Project ID. defaults to None.
```
Comment (cben) [inline lib/ansible/modules/remote_management/manageiq/manageiq_provider.py]: Please mention this is for GCE eg. `description: Google Compute Engine project id`.
Without context, "project id" means nothing...

Comment (snecklifter) [reply lib/ansible/modules/remote_management/manageiq/manageiq_provider.py]: done.

**Comment by yaacov** [comment]
shipit 
again :-)

**Comment by snecklifter** [comment]
Updated auth_key example to clarify this is the field for the google json key.

**Comment by yaacov** [comment]
shipit
againx2  :-)

**Comment by cben** [comment]
:+1:
shipit

**Pull request merged**